crates/pallets/starknet/src/offchain_worker/mod.rs

```diff
@@ -16,20 +16,24 @@ use crate::{Config, Pallet, ETHEREUM_EXECUTION_RPC};
 pub const LAST_FINALIZED_BLOCK_QUERY: &str =
     r#"{"jsonrpc": "2.0", "method": "eth_getBlockByNumber", "params": ["finalized", true], "id": 0}"#;
 
+pub const LAST_GAS_PRICE_QUERY: &str = r#"{"jsonrpc": "2.0", "method": "eth_gasPrice", "params": [], "id": 1}"#;
+
 impl<T: Config> Pallet<T> {
     /// Fetches L1 messages and execute them.
     /// This function is called by the offchain worker.
     /// It is executed in a separate thread.
     /// # Returns
     /// The result of the offchain worker execution.
     pub(crate) fn process_l1_messages() -> Result<(), OffchainWorkerError> {
-        // Get the last known block from storage.
-        let last_known_eth_block = Self::last_known_eth_block().ok_or(OffchainWorkerError::NoLastKnownEthBlock)?;
         // Query L1 for the last finalized block.
         let raw_body = query_eth(LAST_FINALIZED_BLOCK_QUERY)?;
         let last_finalized_block: u64 = from_slice::<EthGetBlockByNumberResponse>(&raw_body)
             .map_err(|_| OffchainWorkerError::SerdeError)?
             .try_into()?;
+
+        // Get the last known block from storage.
+        let last_known_eth_block = Self::last_known_eth_block().ok_or(OffchainWorkerError::NoLastKnownEthBlock)?;
```

Comment:
why change this ?

Reply:
I just changed this to query first because this always returned `NoLastKnownEthBlock` though the rpc_url was wrong.